<html>
<div style="
background:linear-gradient(135deg, #1b1d01, #081d85);
padding:35px;
border-radius:15px;
color:white;
margin-bottom:35px;
">

<h2 style="margin:0;">
Integrated Bioinformatics Analysis of APOE: From Gene Annotation to Comparative Genomics
</h2>

<p style="margin-top:12px;font-size:17px;">
Bioinformatics & Computational Biology Project
</p>

<p style="margin-top:20px;
        font-size:16px;
        opacity:.9;">
Sahel Sepehri • June - July 2026
</p>

</div>
</html>

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Introduction
</h2>

</div>

>- `Apolipoprotein E (APOE)` is a protein-coding gene located on chromosome `19` that plays a key role in lipid transport, cholesterol metabolism, and neuronal function. Genetic variation in APOE is strongly associated with several human diseases, particularly late-onset Alzheimer's disease, making it one of the most extensively studied genes in human genetics.
>- Advances in bioinformatics enable comprehensive exploration of genes using publicly available databases and computational tools. By integrating sequence retrieval, comparative genomics, protein characterization, structural annotation, and clinical variant analysis, it is possible to investigate multiple biological aspects of a gene within a reproducible workflow.
>- This project presents an end-to-end bioinformatics analysis of the human APOE gene, combining information from NCBI, UniProt, ClinVar, BLASTp, MAFFT, and additional bioinformatics resources to examine its sequence characteristics, evolutionary conservation, protein properties, structural features, and clinically relevant variants.

>**Project Objectives:**
>
>This notebook aims to:
>- Retrieve curated APOE gene information from public databases.
>- Obtain reference mRNA and protein sequences programmatically using Biopython.
>- Analyze nucleotide composition and GC content.
>- Characterize protein physicochemical properties using ProtParam.
>- Explore protein domains and functional features using UniProt.
>- Investigate clinically relevant variants using ClinVar.
>- Compare APOE protein sequences across representative vertebrate species using BLASTp.
>- Perform multiple sequence alignment using MAFFT.
>- Construct and interpret a phylogenetic tree to assess evolutionary relationships.

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Project Workflow
</h2>

</div>

```text
NCBI Gene
   ↓
Biopython Retrieval
   ↓
mRNA Analysis
   ↓
Protein Analysis
   ↓
ProtParam Analysis
   ↓
UniProt Functional Features
   ↓
ClinVar Variant Analysis
   ↓
BLASTp Homology Search
   ↓
Multiple Sequence Alignment (MAFFT)
   ↓
Phylogenetic Analysis
   ↓
Biological Interpretation
```

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Import Required Libraries
</h2>

</div>


In [1]:
import pandas as pd

import matplotlib.pyplot as plt

from Bio import Entrez                      # Entrez can actually handles messages (communication) between our computer and the NCBI database.

from Bio import SeqIO                       # SeqIO reads and writes biological sequence file formats like FASTA.

from Bio.SeqUtils import ProtParam          # ProtParam analyzes protein properties (like MW, pI, instability)

from Bio.SeqUtils.ProtParam import ProteinAnalysis   # Calculates physicochemical properties from protein sequences.

from collections import Counter             # Counts the frequency of sequence elements (e.g., nucleotides or amino acids).
 
Entrez.email = "sahelsepehri99@gmail.com"   # We sould tell NCBI who we are so we won't get blocked.


---

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Biological and Database Exploration
</h2>

</div>


> The APOE gene was initially explored using the `NCBI` Gene database to obtain curated biological and genomic information, including its official nomenclature, chromosomal location, gene structure, and functional annotation.
> To provide visual context, screenshots of the APOE gene summary page and genomic region are included. Key gene annotations were then organized into a Python dictionary and converted into a pandas DataFrame, creating a concise summary for subsequent analyses.

<style>
.figure img{
    width:60%;
    display:block;
    margin:auto;
    border-radius:10px;
}
</style>

<div class="figure">

<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/summary2.png?raw=true">
<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/Genomic%20context.png?raw=true">

</div>

In [2]:
apoe_human_info = {
    "Gene Symbol" : "APOE",
    "Official Full Name" : "apolipoprotein E",
    "NCBI Gene ID" : "348",
    "Gene type" : "protein coding",
    "Ensembl ID" : "ENSG00000130203",
    "HGNC ID" : "HGNC:613",
    "Other Names" : "AD2, LPG, APO-E, ApoE4, LDLCQ5",
    "Organism" : "Homo sapiens",
    "Protein Isoforms" : "5",
    "Transcripts Counts" : "5",
    "Chromosome": "19",
    "Location": "19q13.32",
    "Exon count": "6",
    "Intron count" : "5"  #Exon - 1
}


df = pd.DataFrame(list(apoe_human_info.items()), columns= ["Identifier Type", "Value"])

df.style


,Identifier Type,Value
0,Gene Symbol,APOE
1,Official Full Name,apolipoprotein E
2,NCBI Gene ID,348
3,Gene type,protein coding
4,Ensembl ID,ENSG00000130203
5,HGNC ID,HGNC:613
6,Other Names,"AD2, LPG, APO-E, ApoE4, LDLCQ5"
7,Organism,Homo sapiens
8,Protein Isoforms,5
9,Transcripts Counts,5


---

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Retrieving APOE Gene Information Programmatically Using Biopython
</h2>

</div>

> After manually exploring the APOE gene in the NCBI Gene database, the same information was retrieved programmatically using Biopython's `Entrez` module.
> Using `Entrez.esearch()`, the human APOE Gene ID was identified, followed by `Entrez.esummary()` to retrieve curated gene annotations directly from NCBI. The extracted information was organized into a pandas DataFrame, providing a reproducible summary of the gene's basic characteristics.

### Gene ID

In [3]:
# Search the NCBI Gene database for the human APOE gene
search_handle = Entrez.esearch(
    db="gene",
    term="APOE[Gene] AND Homo sapiens[Organism]"
)

# Read the search results and extract the Gene ID
search_record = Entrez.read(search_handle)
gene_id = search_record["IdList"][0]

print(f"APOE Gene ID: {gene_id}")

APOE Gene ID: 348



### Gene Summary

In [4]:
# Retrieve the gene summary using the Gene ID
handle = Entrez.esummary(
    db="gene",
    id=gene_id,
    retmode="xml"
)

summary = Entrez.read(handle)

# Extract the first document summary
doc = summary["DocumentSummarySet"]["DocumentSummary"][0]

# Store selected annotations in a dictionary
data = {
    "Gene Name": doc["Name"],
    "Description": doc["Description"],
    "Chromosome": doc["Chromosome"],
    "Map Location": doc["MapLocation"],
    "Aliases": doc["OtherAliases"]
}

# Convert the dictionary into a DataFrame
df = pd.DataFrame([data])

# Display the table
df

,Gene Name,Description,Chromosome,Map Location,Aliases
0,APOE,apolipoprotein E,19,19q13.32,"AD2, APO-E, ApoE4, LDLCQ5, LPG"


---

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
APOE mRNA Sequence Retrieval and Nucleotide Analysis
</h2>

</div>

<div style="
background: #83b7d0;
border-left:6px solid  #94c1e2;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h3 style="margin:0;color: #0F172A;">
Programmatic Retrieval of APOE mRNA Sequence
</h3>

</div>




> The human **APOE mRNA sequence** was retrieved programmatically from the NCBI Nucleotide database using **Biopython**.
> The RefSeq accession **`NM_000041.4`** was used as the reference APOE transcript.
> The sequence was obtained using `Entrez.efetch()` and parsed with the `SeqIO` module for downstream analysis.
> 
> The first 100 nucleotides were displayed as an initial quality check.
> 
> Although NM_000041.4 represents an mRNA transcript, RefSeq nucleotide records are stored using the DNA alphabet (T instead of U). The sequence was converted into RNA notation using transcription when required.

In [5]:
# Entrez.efetch() and SeqIO parsing code
# Retrieve the reference APOE mRNA sequence from the NCBI Nucleotide database

# Fetch means "to go and get something or someone and bring them back"
# We ask NCBI to fetch a specific mRNA ID and save the connection handle
# id="NM_000041.4" -> is the RefSeq accession of the human APOE mRNA transcript
# rettype="fasta" -> returns the nucleotide sequence in FASTA format.
# retmode = "text" -> returns a simple text format (FASTA) instead of XML 

handle = Entrez.efetch(
    db="nucleotide",
    id="NM_000041.4",
    rettype="fasta",
    retmode="text"
)

# Read the FASTA record using SeqIO
mrna_record = SeqIO.read(handle, "fasta")
handle.close()

# Extract the nucleotide sequence
mrna_seq = mrna_record.seq

# Display sequence description
print(mrna_record.description)

# Display the first 100 nucleotides
print("\nFirst 100 nucleotides:")
print("- Reference nucleotide sequence (DNA notation, T):")
print(mrna_seq[:100])



# Convert the DNA-style nucleotide sequence (T) into an RNA sequence (U)
rna_seq = mrna_seq.transcribe()


print("\n- RNA sequence after transcription (U):")
print(rna_seq[:100])

NM_000041.4 Homo sapiens apolipoprotein E (APOE), transcript variant 2, mRNA

First 100 nucleotides:
- Reference nucleotide sequence (DNA notation, T):
CTACTCAGCCCCAGCGGAGGTGAAGGACGTCCTTCCCCAGGAGCCGACTGGCCAATCACAGGCAGGAAGATGAAGGTTCTGTGGGCTGCGTTGCTGGTCA

- RNA sequence after transcription (U):
CUACUCAGCCCCAGCGGAGGUGAAGGACGUCCUUCCCCAGGAGCCGACUGGCCAAUCACAGGCAGGAAGAUGAAGGUUCUGUGGGCUGCGUUGCUGGUCA


</div>

<div style="
background: #83b7d0;
border-left:6px solid  #94c1e2;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h3 style="margin:0;color: #0F172A;">
Nucleotide Composition and GC Content Analysis
</h3>

</div>

> **Objective:**
> To characterize the `nucleotide composition` and calculate the `GC content` of the APOE transcript.
> 
>**Method:**
> The frequency of each nucleotide (**`A, C, G, and T`**) was calculated, and GC content was determined as the `percentage of guanine and cytosine nucleotides` within the sequence.


In [6]:
# Nucleotide counting and GC content calculation code

# Convert sequence to uppercase
sequence = str(mrna_record.seq).upper()

# Sequence length
total = len(sequence)

# Count nucleotides
a_count = sequence.count("A")
c_count = sequence.count("C")
g_count = sequence.count("G")
t_count = sequence.count("T")

# Calculate GC content
gc_content = (g_count + c_count) / total * 100

# Create DataFrame
df_nt = pd.DataFrame({
    "Nucleotide": ["A", "C", "G", "T"],
    "Count": [a_count, c_count, g_count, t_count],
    "Percentage (%)": [
        round(a_count / total * 100, 2),
        round(c_count / total * 100, 2),
        round(g_count / total * 100, 2),
        round(t_count / total * 100, 2)
    ]
})

# Display results
print(f"Sequence Length: {total} nt")
print(f"GC Content: {gc_content:.2f}%\n")

df_nt

Sequence Length: 1166 nt
GC Content: 69.04%



,Nucleotide,Count,Percentage (%)
0,A,211,18.10
1,C,373,31.99
2,G,432,37.05
3,T,150,12.86


### > Results

>The APOE transcript contains:
>- **Sequence length:** 1166 nucleotides
>- **GC content:** 69.04%
> The nucleotide composition is summarized in the table above.

### > Interpretation

> - The nucleotide composition provides a basic characterization of the APOE transcript sequence.
> - The relatively high GC content indicates a `GC-rich sequence`, which may contribute to `increased nucleotide stability` due to stronger G–C base pairing.
> - This analysis represents an initial quality assessment before downstream analyses, including translation, protein characterization, and evolutionary comparison.

---

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
APOE Protein Sequence Retrieval and Characterization
</h2>

</div>

<div style="
background: #83b7d0;
border-left:6px solid  #94c1e2;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h3 style="margin:0;color: #0F172A;">
APOE Protein Information
</h3>

</div>

> - The human APOE protein sequence was obtained from the NCBI Protein database using the RefSeq accession **`NP_000032.1`**.
> - NP_000032.1 corresponds to the **`apolipoprotein E isoform b precursor`**, encoded by transcript variant 2 (**NM_000041.4**).

<style>
.figure img{
    width:60%;
    display:block;
    margin:auto;
    border-radius:10px;
}
</style>

<div class="figure">

<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/transcripts_proteins.png?raw=true">
</div>

| Feature    | Description                          |
| ---------- | ------------------------------------ |
| Protein    | apolipoprotein E isoform b precursor |
| RefSeq ID  | NP_000032.1                          |
| Length     | 317 amino acids                      |
| Organism   | Homo sapiens                         |
| Transcript | NM_000041.4                          |


<div style="
background: #83b7d0;
border-left:6px solid  #94c1e2;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h3 style="margin:0;color: #0F172A;">
Programmatic Retrieval of APOE Protein Sequence
</h3>

</div>


> - The APOE protein sequence was retrieved programmatically from NCBI using `Biopython`.
> - The sequence was fetched using the RefSeq accession **`NP_000032.1`**, parsed using `SeqIO`, and saved in FASTA format for downstream analysis.

In [7]:
handle = Entrez.efetch(             # Fetch APOE protein sequence from NCBI using its RefSeq ID
    db="protein",
    id="NP_000032.1",
    rettype="fasta",
    retmode="text"
    )

protein_record = SeqIO.read(handle, "fasta")    # Read the sequence and close connection
handle.close()

# Display sequence info
print(protein_record.description)
print("\nFirst 100 characters of FASTA:")
print(protein_record.seq[:100])
print("Protein length:", len(protein_record.seq), "a.a")


# Save to file
SeqIO.write(protein_record, "APOE_protein.fasta", "fasta")
print("\nSaved as APOE_protein.fasta")


NP_000032.1 apolipoprotein E isoform b precursor [Homo sapiens]

First 100 characters of FASTA:
MKVLWAALLVTFLAGCQAKVEQAVETEPEPELRQQTEWQSGQRWELALGRFWDYLRWVQTLSEQVQEELLSSQVTQELRALMDETMKELKAYKSELEEQL
Protein length: 317 a.a

Saved as APOE_protein.fasta


<style>
.figure img{
    width:60%;
    display:block;
    margin:auto;
    border-radius:10px;
}
</style>

<div class="figure">

<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/protein_seq.png?raw=true">
</div>

<div style="
background: #83b7d0;
border-left:6px solid  #94c1e2;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h3 style="margin:0;color: #0F172A;">
Protein Functional Features (UniProt)
</h3>

</div>

> UniProt provides manually curated annotations describing the structural and functional features of proteins.
> The APOE protein sequence was further explored using the UniProt Knowledgebase (Swiss-Prot) to identify important functional regions, processing events, binding sites, and post-translational modifications.

<style>
.figure img{
    width:60%;
    display:block;
    margin:auto;
    border-radius:10px;
}
</style>

<div class="figure">

<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/APOE-UNIPROT-features.png?raw=true">
</div>

The annotated features obtained from UniProt complement the sequence-based ProtParam analysis by providing experimentally curated functional information.

**Figure X.** UniProt feature viewer showing the annotated functional regions, processing events, binding sites, and post-translational modifications of the human APOE protein (UniProt: P02649).

| Feature                 | Position | Description                      |
| ----------------------- | -------: | -------------------------------- |
| Signal peptide          |     1–18 | Directs secretion of APOE        |
| Mature chain            |   19–317 | Functional extracellular protein |
| Receptor-binding region |  136–150 | Interaction with LDL receptor    |
| Lipid-binding region    |  244–272 | Lipid transport                  |


### Interpretation

The UniProt annotations indicate that APOE contains several well-defined functional regions required for its biological activity.

The N-terminal signal peptide directs protein secretion, whereas the receptor-binding and lipid-binding regions are essential for lipoprotein metabolism and receptor interactions.

| PTM            | Position | Function             |
| -------------- | -------: | -------------------- |
| Disulfide bond |       xx | Structural stability |
| Glycosylation  |       xx | Protein maturation   |

| Binding site | Ligand  |
| ------------ | ------- |
| Heparin      | Binding |
| Lipid        | Binding |


<div style="
background: #83b7d0;
border-left:6px solid  #94c1e2;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h3 style="margin:0;color: #0F172A;">
Protein Physicochemical Analysis (ProtParam)
</h3>

</div>

>**Objective:**
>To characterize the physicochemical properties of the human APOE protein based on its amino acid sequence.
>
>**Method:**
> The APOE amino acid sequence was analyzed using the **`ProtParam module in Biopython`**.
>
> The following properties were calculated:
>- Amino acid composition
>- Molecular weight
>- Theoretical isoelectric point (pI)
>- Instability index
>- Aromaticity
>- GRAVY score (hydrophobicity)
>- Secondary structure fractions

In [8]:
# APOE Protein Properties (ProtParam Analysis)
# Create a ProtParam object from the protein sequence
my_protein = ProtParam.ProteinAnalysis(str(protein_record.seq))

<div style="
background: #e3f0f1;
border-left:6px solid  #dde1e0;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h5 style="margin:0;color: #0F172A;">
1. Amino Acid Composition
</h5>

</div>

> The frequency and percentage of each amino acid were calculated to describe the sequence composition of APOE.

In [9]:
# Load the protein sequence
handle = Entrez.efetch(db="protein", id="NP_000032.1", rettype="fasta", retmode="text")
protein_record = SeqIO.read(handle, "fasta")
handle.close()

# Create ProtParam object for protein analysis
my_protein = ProtParam.ProteinAnalysis(str(protein_record.seq))

# Count amino acid frequencies
sequence = str(protein_record.seq)
aa_counts = Counter(sequence)
total_aa = sum(aa_counts.values())

# Amino acid dictionary: 1-letter code → (Full Name, 3-letter code)
aa_names = {
    'A': ('Alanine', 'Ala'), 'R': ('Arginine', 'Arg'), 'N': ('Asparagine', 'Asn'),
    'D': ('Aspartic Acid', 'Asp'), 'C': ('Cysteine', 'Cys'), 'Q': ('Glutamine', 'Gln'),
    'E': ('Glutamic Acid', 'Glu'), 'G': ('Glycine', 'Gly'), 'H': ('Histidine', 'His'),
    'I': ('Isoleucine', 'Ile'), 'L': ('Leucine', 'Leu'), 'K': ('Lysine', 'Lys'),
    'M': ('Methionine', 'Met'), 'F': ('Phenylalanine', 'Phe'), 'P': ('Proline', 'Pro'),
    'S': ('Serine', 'Ser'), 'T': ('Threonine', 'Thr'), 'W': ('Tryptophan', 'Trp'),
    'Y': ('Tyrosine', 'Tyr'), 'V': ('Valine', 'Val')
}

# Build composition table
data = []
for aa, count in aa_counts.most_common():
    if aa in aa_names:
        full_name, three_letter = aa_names[aa]
        percentage = (count / total_aa) * 100
        data.append([aa, three_letter, full_name, count, round(percentage, 2)])
        
        
        
# Display as DataFrame
# The table below summarizes the percentage of each amino acid in the APOE protein sequence.
df_aa = pd.DataFrame(data, columns=['1-Letter', '3-Letter', 'Name', 'Count', '%'])
print(df_aa)


   1-Letter 3-Letter           Name  Count      %
0         L      Leu        Leucine     41  12.93
1         E      Glu  Glutamic Acid     40  12.62
2         A      Ala        Alanine     39  12.30
3         R      Arg       Arginine     34  10.73
4         Q      Gln      Glutamine     32  10.09
5         V      Val         Valine     24   7.57
6         G      Gly        Glycine     18   5.68
7         S      Ser         Serine     14   4.42
8         K      Lys         Lysine     13   4.10
9         T      Thr      Threonine     12   3.79
10        D      Asp  Aspartic Acid     11   3.47
11        M      Met     Methionine      8   2.52
12        W      Trp     Tryptophan      8   2.52
13        P      Pro        Proline      8   2.52
14        F      Phe  Phenylalanine      4   1.26
15        Y      Tyr       Tyrosine      4   1.26
16        C      Cys       Cysteine      2   0.63
17        H      His      Histidine      2   0.63
18        I      Ile     Isoleucine      2   0.63


Interpretation

> The amino acid composition provides a general overview of the APOE protein sequence and serves as the basis for subsequent physicochemical analyses.

---

<div style="
background: #e3f0f1;
border-left:6px solid  #dde1e0;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h5 style="margin:0;color: #0F172A;">
2. Molecular Weight
</h5>

</div>

In [10]:
# Calculate the molecular weight of the APOE protein
mw = my_protein.molecular_weight()

print(f"Molecular weight: {mw:.2f} Da")

Molecular weight: 36153.66 Da


Interpretation:
>The predicted molecular weight of APOE is approximately **36.15 kDa**, which is consistent with the expected size of the mature human protein.

---

<div style="
background: #e3f0f1;
border-left:6px solid  #dde1e0;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h5 style="margin:0;color: #0F172A;">
3. Theoretical Isoelectric Point (pI)
</h5>

</div>

In [11]:
# Calculate the theoretical isoelectric point (pI)
pi = my_protein.isoelectric_point()

print(f"Isoelectric point (pI): {pi:.2f}")

Isoelectric point (pI): 5.65


Interpretation

> The theoretical pI of **`5.65`** indicates that APOE is expected to carry a `net negative charge` under physiological conditions, which may influence its interactions with lipids and receptors.

---

<div style="
background: #e3f0f1;
border-left:6px solid  #dde1e0;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h5 style="margin:0;color: #0F172A;">
4. Instability index 
</h5>

</div>

In [12]:
#Instability index (below 40 = stable, above 40 = unstable)
instability = my_protein.instability_index()
print(f"Instability index: {instability:.2f}")
if instability < 40:
    print("Protein is predicted to be Stable.")
else:
    print("Protein is predicted to be Unstable.")

Instability index: 44.88
Protein is predicted to be Unstable.


Interpretation

> The predicted instability index (**`44.88`**) is slightly above the ProtParam stability threshold, suggesting potential `instability` based solely on amino acid composition. This computational prediction does not necessarily reflect protein stability under physiological conditions.

---

<div style="
background: #e3f0f1;
border-left:6px solid  #dde1e0;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h5 style="margin:0;color: #0F172A;">
5. Aromaticity
</h5>

</div>

In [13]:
# Calculate the aromaticity of the APOE protein
aromaticity = my_protein.aromaticity()


print(f"Aromaticity: {aromaticity:.2f}")

Aromaticity: 0.05


Interpretation

> The aromaticity value (**`0.05`**) indicates that `aromatic amino acids constitute a relatively small proportion of the APOE protein sequence.

---

<div style="
background: #e3f0f1;
border-left:6px solid  #dde1e0;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h5 style="margin:0;color: #0F172A;">
6. GRAVY (Grand Average of Hydropathy)
</h5>

</div>


In [14]:
# Calculate the GRAVY score
gravy = my_protein.gravy()


print(f"GRAVY (hydrophobicity): {gravy:.2f}")

if gravy > 0:
    print("Hydrophobic protein")
else:
    print("Hydrophilic protein")

GRAVY (hydrophobicity): -0.60
Hydrophilic protein


Interpretation

> The negative GRAVY score (**`−0.60`**) indicates that APOE is `predominantly hydrophilic`, which is consistent with its function as a soluble plasma apolipoprotein involved in lipid transport.

---

<div style="
background: #e3f0f1;
border-left:6px solid  #dde1e0;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h5 style="margin:0;color: #0F172A;">
7. Secondary Structure Fraction
</h5>

</div>


In [15]:
sec_structure = my_protein.secondary_structure_fraction ()
print(f"Helix fraction: {sec_structure[0]:.3f} ({sec_structure[0]*100:.1f}%)")
print(f"Turn fraction:  {sec_structure[1]:.3f} ({sec_structure[1]*100:.1f}%)")
print(f"Sheet fraction: {sec_structure[2]:.3f} ({sec_structure[2]*100:.1f}%)")


Helix fraction: 0.445 (44.5%)
Turn fraction:  0.164 (16.4%)
Sheet fraction: 0.300 (30.0%)


Interpretation

> ProtParam predicts that APOE contains a high proportion of **`α-helices`** together with moderate **`β-sheet`** content.
> These values provide a general description of structural composition based on amino acid sequence and should not be interpreted as an experimentally determined three-dimensional structure.

---

<div style="
background: #e3f0f1;
border-left:6px solid  #dde1e0;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h5 style="margin:0;color: #0F172A;">
Protein Characterization Summary
</h5>

</div>


In [16]:
sequence = str(protein_record.seq)
my_protein = ProteinAnalysis(sequence)

# Compute properties
mw = my_protein.molecular_weight()
pI = my_protein.isoelectric_point()
gravy = my_protein.gravy()
instability = my_protein.instability_index()
aromaticity = my_protein.aromaticity()
helix, turn, sheet = my_protein.secondary_structure_fraction()

# Build final table
data = [
    ["Molecular Weight (Da)", mw],
    ["Isoelectric Point (pI)", pI],
    ["GRAVY (Hydrophobicity)", gravy],
    ["Instability Index", instability],
    ["Aromaticity", aromaticity],
    ["Helix Fraction", helix],
    ["Turn Fraction", turn],
    ["Sheet Fraction", sheet]
]

# Create summary table of ProtParam results
df = pd.DataFrame(data, columns=["Property", "Value"])

# Round numerical values
df["Value"] = df["Value"].round(2)
# Display table
display(df)

,Property,Value
0,Molecular Weight (Da),36153.66
1,Isoelectric Point (pI),5.65
2,GRAVY (Hydrophobicity),-0.60
3,Instability Index,44.88
4,Aromaticity,0.05
5,Helix Fraction,0.44
6,Turn Fraction,0.16
7,Sheet Fraction,0.30


### Overall Interpretation

> ProtParam analysis indicates that APOE is a `predominantly hydrophilic protein with a slightly acidic isoelectric point and a mainly α-helical structure. Although the instability index is slightly above the ProtParam threshold, this prediction is sequence-based and does not necessarily represent protein behavior in vivo.
> Overall, these results provide a computational characterization of the APOE protein that is consistent with its known biological role in lipid transport and receptor-mediated metabolism.

---

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Clinical Variant Analysis (ClinVar)
</h2>

</div>

Introduction

ClinVar is a publicly available database maintained by the NCBI that provides information on the relationships between human genetic variants and clinical phenotypes.

In this section, ClinVar was used to investigate clinically relevant variants reported for the APOE gene.

Objective

To identify clinically relevant APOE variants and summarize their reported disease associations.

Method

The APOE gene was searched in the ClinVar database. Representative pathogenic and risk-associated variants were reviewed together with their reported clinical significance.

| Variant  | Protein Change | Clinical Significance | Associated Condition           |
| -------- | -------------- | --------------------- | ------------------------------ |
| rs429358 | p.Cys130Arg    | Risk factor           | Late-onset Alzheimer's disease |
| rs7412   | p.Arg176Cys    | Risk factor           | Late-onset Alzheimer's disease |



| Variant  | Protein Change | Clinical Significance | Associated Condition                   |
| -------- | -------------- | --------------------- | -------------------------------------- |
| c.388T>C | p.Cys130Arg    | Pathogenic            | Familial type III hyperlipoproteinemia |
| c.526C>T | p.Arg176Cys    | Pathogenic            | Familial type III hyperlipoproteinemia |
| c.548G>C | p.Gly183Ala    | Likely pathogenic     | Familial type III hyperlipoproteinemia |
| c.526C>T | p.Arg176Cys    | Drug response         | Atorvastatin response                  |


### Interpretation

ClinVar analysis identified several clinically relevant APOE variants associated with lipid metabolism disorders and Alzheimer's disease susceptibility.

The variants **rs429358** and **rs7412** define the major APOE isoforms (**ε2, ε3, and ε4**), which are well-established genetic determinants of Alzheimer's disease risk.

Several pathogenic variants were also associated with **familial type III hyperlipoproteinemia**, highlighting the important role of APOE in lipid metabolism. In addition, one variant was reported to influence the response to atorvastatin, demonstrating the pharmacogenomic relevance of APOE.

### Summary

Overall, ClinVar results demonstrate that APOE harbors multiple clinically significant variants associated with human disease, emphasizing the importance of this gene in both lipid metabolism and neurodegenerative disorders.


---

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
BLASTp Homology Analysis
</h2>

</div>

> - The human APOE protein sequence was compared with homologous protein sequences from representative vertebrate species using the `NCBI BLASTp tool`. BLASTp identifies proteins with similar amino acid sequences and provides quantitative measures of sequence similarity, allowing evolutionary conservation to be evaluated across species.
>- This analysis was performed to identify APOE orthologs for subsequent multiple sequence alignment (MAFFT) and phylogenetic analysis.


> **Objective**
> - Identify APOE orthologs from representative vertebrate species.
> - Evaluate sequence similarity and evolutionary conservation.
> - Select representative protein sequences for multiple sequence alignment and phylogenetic analysis.

> **Method**
> - The human APOE protein sequence (RefSeq: NP_000032.1) was queried against the NCBI refseq protein database using BLASTp.
> - One representative APOE ortholog was selected from each species.
> - Ten vertebrate species spanning different evolutionary distances were included for comparative analysis.

> **Selected Species**
>- **Primates:** Human, Chimpanzee, Gorilla, Orangutan, and Rhesus macaque
>- **Other mammals:** Mouse, Rat, Dog, and Cattle
>- **Teleost fish:** Zebrafish

```text
Human
│
├── Chimpanzee
├── Gorilla
├── Orangutan
├── Rhesus macaque
│
├── Mouse
├── Rat
├── Dog
├── Cattle
│
└── Zebrafish
```

<style>
.figure img{
    width:60%;
    display:block;
    margin:auto;
    border-radius:10px;
}
</style>

<div class="figure">

<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/BLASTp.png?raw=true">
</div>

<style>
.figure img{
    width:60%;
    display:block;
    margin:auto;
    border-radius:10px;
}
</style>

<div class="figure">

<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/BLASTp-Results.png?raw=true">
</div>

### Results

**Understanding BLAST Metrics**

| Term               | Description                                                                                  |
| ------------------ | -------------------------------------------------------------------------------------------- |
| Identity (%)       | Percentage of identical amino acids between the human APOE sequence and the matched protein. |
| Query Coverage (%) | Percentage of the human APOE sequence aligned with the matched protein.                      |
| E-value            | Statistical significance of the alignment. Lower values indicate more significant matches.   |
| Protein Length     | Length of the matched protein in amino acids.                                                |


   **BLASTp results of the human APOE protein against representative vertebrate species:**
| Species | Scientific Name | Identity (%) | Query Coverage (%) | E-value | Protein Length (aa) |
|---------|-----------------|-------------:|-------------------:|--------:|--------------------:|
| Human | *Homo sapiens* | 100.00 | 100 | 0 | 317 |
| Chimpanzee | *Pan troglodytes* | 97.16 | 100 | 0 | 317 |
| Western lowland gorilla | *Gorilla gorilla gorilla* | 97.48 | 100 | 0 | 344 |
| Sumatran orangutan | *Pongo abelii* | 96.53 | 100 | 0 | 344 |
| Indochinese rhesus macaque | *Macaca mulatta* | 93.69 | 100 | 0 | 317 |
| House mouse | *Mus musculus* | 70.85 | 100 | 9e-156 | 311 |
| Domestic cattle | *Bos taurus* | 72.28 | 96 | 1e-147 | 316 |
| Dog | *Canis lupus familiaris* | 68.42 | 99 | 3e-145 | 323 |
| Norway rat | *Rattus norvegicus* | 69.71 | 96 | 3e-143 | 312 |
| Zebrafish | *Danio rerio* | 26.25 | 94 | 9e-143 | 269 |

### Interpretation

> The BLASTp analysis demonstrated that APOE is `highly conserved among vertebrates`.
> - Primate APOE proteins showed very high sequence identity (>93%) with complete query coverage, indicating strong evolutionary conservation.
> - Other mammals exhibited moderate sequence identity (approximately 68–72%) while maintaining high query coverage.
> - Zebrafish displayed substantially lower sequence identity (26.25%), reflecting its greater evolutionary distance from mammals.
> 
> Overall, these results indicate that `APOE has remained highly conserved throughout vertebrate evolution`, supporting its fundamental biological role.
> The selected orthologs were subsequently used for multiple sequence alignment (MAFFT) and phylogenetic analysis.

---

<div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Multiple Sequence Alignment (MAFFT)
</h2>

</div>


> - Multiple sequence alignment (MSA) was performed to investigate the evolutionary conservation of the APOE protein across representative vertebrate species.
> - By aligning homologous protein sequences, conserved amino acid residues, variable regions, and potential functionally important sites can be identified.
> - This analysis complements the BLASTp results by providing residue-level comparisons rather than overall sequence similarity.
>
> ### Dataset
> Protein sequences corresponding to the selected BLASTp hits were downloaded from the NCBI RefSeq Protein database in FASTA format.
> The dataset included ten representative vertebrate species:
>- Human
>- Chimpanzee
>- Gorilla
>- Orangutan
>- Rhesus macaque
>- Mouse
>- Rat
>- Dog
>- Cattle
>- Zebrafish
>
>
> ### Alignment Method
> The protein sequences were aligned using the MAFFT online server with the default parameters.
> The resulting alignment was visualized to examine conserved and variable amino acid positions across species.

<style>
.figure img{
    width:60%;
    display:block;
    margin:auto;
    border-radius:10px;
}
</style>

<div class="figure">

<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/MAFFT.png?raw=true">

</div>

> Multiple sequence alignment of APOE protein sequences from ten representative vertebrate species generated using MAFFT.

[MAFFT MSA Viewer](https://mafft.cbrc.jp/msaviewer/?LoadURL=/alignment/server/spool/_out.2607161546mafftweba1835113jlBESoRV9qMcHKn5hu61blsfnormal.pir&Pos=281,0,45&ColorScheme=Miyata#)

> ### Interpretation
> - The alignment demonstrates a high level of sequence conservation among primates, particularly across the central and C-terminal regions of the APOE protein.
> - Greater sequence variation is observed in more evolutionarily distant species, especially zebrafish, which contains numerous amino acid substitutions and insertion/deletion events.
> - Despite these differences, several conserved residues remain preserved across all vertebrates, suggesting that important structural and functional regions of APOE have been maintained throughout evolution.
> - Overall, the MSA supports the BLASTp analysis and provides additional evidence that APOE is highly conserved among mammals, especially primates.

---

<div style="
background: #e7e9ea;
border-left:6px solid #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color:#0F172A;">
Phylogenetic Analysis
</h2>

</div>


> - A phylogenetic analysis was performed to investigate the evolutionary relationships among APOE proteins from representative vertebrate species.
> - The phylogenetic tree was constructed using the aligned protein sequences generated from the MAFFT multiple sequence alignment.
> - This analysis was performed to determine whether the clustering pattern of APOE proteins reflects the evolutionary relationships among the selected species.
> 
> **Dataset**
> The phylogenetic analysis was based on the same set of APOE protein sequences used for multiple sequence alignment.
> 
> The dataset included ten representative vertebrate species:
>- Human
>- Chimpanzee
>- Gorilla
>- Orangutan
>- Rhesus macaque
>- Mouse
>- Rat
>- Dog
>- Cattle
>- Zebrafish

> **Tree Construction**
>- The APOE protein sequences were first aligned using the `MAFFT` online server. The resulting multiple sequence alignment was subsequently used to infer a phylogenetic tree using the `Neighbor-Joining (NJ`) method with the default parameters provided by the MAFFT phylogeny module. The generated tree was visualized using the integrated `Phylo.io/Archaeopteryx viewer`
> 
> **Output**
> - Phylogenetic tree showing evolutionary relationships among APOE proteins.
> - Clustering pattern of APOE homologs across vertebrate species.
> - Visualization of evolutionary conservation and divergence among species.

<style>
.figure img{
    width:60%;
    display:block;
    margin:auto;
    border-radius:10px;
}
</style>

<div class="figure">

<img src="https://github.com/sahelsepehri/APOE-Comparative-Genomics/blob/main/Images/archaeopteryx_js.png?raw=true">

</div>

> - [View Tree on Phylo.io](https://mafft.cbrc.jp/alignment/server/spool/_phyloio.26071615541869376.html)
> - [View Tree on Archaeopteryx.js](https://mafft.cbrc.jp/alignment/server/spool/_aptx.26071615541869376.html)
> - [View on EMBL-EBI, Simple_Phylogeny](https://www.ebi.ac.uk/jdispatcher/phylogeny/simple_phylogeny/summary?jobId=simple_phylogeny-I20260716-081231-0145-37075152-p1m)


> ### Tree Interpretation 
> The phylogenetic tree showed a clustering pattern consistent with the evolutionary relationships among the selected vertebrate species. APOE proteins from primates grouped together, with human, chimpanzee, gorilla, orangutan, and rhesus macaque sequences forming a closely related cluster. Mouse and rat APOE sequences also clustered together, consistent with their close evolutionary relationship. Other mammalian species, including dog and cattle, occupied separate branches within the mammalian group, while the zebrafish sequence showed the greatest divergence from the mammalian APOE proteins. These results suggest that APOE has been strongly conserved during mammalian evolution, while more sequence differences have accumulated in distantly related vertebrates.



 <div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Final Conclusion
</h2>

</div>

 <div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Limitations
</h2>

</div>



 <div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
Future Work
</h2>

</div>




 <div style="
background: #e7e9ea;
border-left:6px solid  #2693e1;
padding:15px 20px;
border-radius:10px;
margin:40px 0 25px 0;
">

<h2 style="margin:0;color: #0F172A;">
References 
</h2>

</div>


- [NCBI](https://www.ncbi.nlm.nih.gov/)
- [Pandas Documentation](https://pandas.pydata.org/docs/index.html)
- [Accessing NCBI’s Entrez databases - biopython](https://biopython.org/docs/latest/Tutorial/chapter_entrez.html)
- [UniProt](https://www.uniprot.org/)
- [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)
- [Analyzing protein sequences with the ProtParam module](https://biopython.org/wiki/ProtParam)
- [BLASTp](https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins)
- [MAFFT version 7](https://mafft.cbrc.jp/alignment/server/index.html)